okhttp/src/main/java/okhttp3/internal/connection/StreamAllocation.java

```diff
@@ -301,6 +301,9 @@ private Socket deallocate(boolean noNewStreams, boolean released, boolean stream
       this.codec = null;
     }
     if (released) {
+      if (connection != null) {
+        eventListener.connectionReleased(call, connection);
```

Comment:
I’m a little anxious about calling user code while holding the single global connection pool lock. Probably safer to call this without the lock